zkvm/src/types.rs

```diff
@@ -165,13 +173,28 @@ impl Commitment {
             Commitment::Closed(x) => Ok(*x),
         }
     }
+
+    fn encode(&self, buf: &mut Vec<u8>) {
+        buf.extend_from_slice(&self.to_point().to_bytes());
+    }
 }
 
 impl CommitmentWitness {
     pub fn to_point(&self) -> CompressedRistretto {
         let gens = PedersenGens::default();
         gens.commit(self.value.into(), self.blinding).compress()
     }
+
+    fn encode(&self, buf: &mut Vec<u8>) {
+        buf.extend_from_slice(&self.to_point().to_bytes());
+    }
+}
+
+impl InputWitness {
+    // TBD: need initializer to construct InputWitness
+    fn destructure(&self) -> (&FrozenContract, UTXO, TxID) {
```

Comment:
maybe we'll need a consuming self version of this later, but now we can replace this method with `encode`:

```rust
pub fn encode(&self, buf: &mut Vec<u8>) {
    buf.extend_from_slice(&self.txid.0);
    self.contract.encode(buf);
}
```